resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/PlainObjectBulkheadAspectExt.java

```diff
@@ -0,0 +1,139 @@
+package io.github.resilience4j.bulkhead.configure;
+
+import io.github.resilience4j.bulkhead.Bulkhead;
+import io.github.resilience4j.bulkhead.BulkheadFullException;
+import io.github.resilience4j.bulkhead.ThreadPoolBulkhead;
+import io.github.resilience4j.bulkhead.internal.ThreadPoolBulkheadAdapter;
+import io.github.resilience4j.timelimiter.TimeLimiter;
+import io.github.resilience4j.timelimiter.TimeLimiterRegistry;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.*;
+import java.util.function.Supplier;
+
+/**
+ * The {@code PlainObjectBulkheadAspectExt} class provides an aspect for managing method executions
+ * that return plain objects while enforcing bulkhead constraints.
+ */
+public class PlainObjectBulkheadAspectExt implements BulkheadAspectExt {
+
+    private static final Logger logger = LoggerFactory.getLogger(PlainObjectBulkheadAspectExt.class);
+    private final TimeLimiterRegistry timeLimiterRegistry;
+    private final ScheduledExecutorService executorService;
+
+    public PlainObjectBulkheadAspectExt(TimeLimiterRegistry timeLimiterRegistry) {
+        this.timeLimiterRegistry = timeLimiterRegistry;
+        this.executorService = Executors.newScheduledThreadPool(Runtime.getRuntime().availableProcessors());
```

Comment:
Have a look here: https://github.com/resilience4j/resilience4j/blob/master/resilience4j-spring/src/main/java/io/github/resilience4j/timelimiter/configure/TimeLimiterAspect.java#L54-L68

The TimeLimiter needs a special `ContextAwareScheduledThreadPoolExecutor` so that context information can be passed from Thread to Thread